HealthVault/Classes/Caching/MHVThingCacheDatabaseProtocol.h

```diff
@@ -19,110 +19,140 @@
+ */
+- (void)createCachedThings:(MHVThingCollection *)things
+                  recordId:(NSString *)recordId
+                completion:(void (^)(NSError *_Nullable error))completion;
+
+/**
+ This method is called AFTER a Thing (or Things) have been successfully updated in HealthVault.
+
+ @param things A collection of existing things to be updated in the cache database.
+ @param recordId the RecordId of the owner of the things.
+ @param completion MUST be envoked when the operation is complete or an error occurs. NSError error a detailed error if the creation process could not be completed.
  */
-- (void)deleteRecord:(NSString *)recordId
-          completion:(void (^)(NSError *_Nullable error))completion;
+- (void)updateCachedThings:(MHVThingCollection *)things
+                  recordId:(NSString *)recordId
+                completion:(void (^)(NSError *_Nullable error))completion;
 
 /**
- Delete Things given an array of IDs
+ This method is called AFTER a Thing (or Things) have been successfully deleted from HealthVault.
  
  @param thingIds the IDs of the things to be deleted
  @param recordId the RecordId of the owner of the things
  @param completion Envoked when the operation is complete
  */
-- (void)deleteThingIds:(NSArray<NSString *> *)thingIds
-              recordId:(NSString *)recordId
-            completion:(void (^)(NSError *_Nullable error))completion;
+- (void)deleteCachedThingsWithThingIds:(NSArray<NSString *> *)thingIds
+                              recordId:(NSString *)recordId
+                            completion:(void (^)(NSError *_Nullable error))completion;
 
 /**
- Update or create things in the cache database for a Thing collection
- If the thingId is found, it will be updated; if not it will be added
+ This method will be called to synchronize Things from HealthVault to the cache database for a given recordId.
  
- @param things collection of Things to be added or updated
- @param recordId the owner record of the Things
- @param lastSequenceNumber the new sequence number to use after updating
- @param completion Envoked when the operation is complete
+ @note the synchronization may include a large number of Things from HealthVault. In this case, the data is split into batches that are processed serially. The batchSequenceNumber and latestSequenceNumber parameters are used to determine if all batches have been processed - Where, batchSequenceNumber represents the greatest sequence number for a given batch, and latestSequenceNumber represents the greatest sequence number for all batches. During initial setup of the cache, or after long periods where a sync did not happen, this method will be called repeatedly until batchSequenceNumber = latestSequenceNumber. It's recommended that batchSequenceNumber be saved so if the sync is interupted the process can be continued.
+ @param things collection of Things to be synchronized.
```

Comment:
I'd note that the thing collection may contain both things to be added, as well as newer versions of things that may already exist and should be updated.